flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -59,6 +59,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
           Literal(str.toInt)
         } else if (str.endsWith("f") | str.endsWith("F")) {
           Literal(str.toFloat)
+        } else if (str.endsWith(".")) {
```

Comment:
Seems a little bit like a hack to me. Ideally, you should design your grammar such that you won't see the token `1.` if it is not a valid number expression.